ezsystems/ezplatform-richtext
# EZP-31642: As an Editor I want be able to edit table borders

| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-31642](https://jira.ez.no/browse/EZP-31642)
| **Bug/Improvement**| yes
| **New feature**    | yes
| **Target version** | 2.1
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

This PR provides a new concept for custom attributes in Online Editor. In some cases, there is no need to use data attributes for them. Table border is a great example, and I guess there might be more cases.

Steps to test it:
1. Just add `border` custom attribute for tables:
    ```
    system:
        admin_group:
            fieldtypes:
                ezrichtext:
                    attributes:
                        table:
                            border: { type: number, default_value: 1 }
    ```

2. Open Online Editor and insert the table

3. Using elements path (at the bottom of the Online Editor) select a table

4. Click "Edit" icon in the table toolbar

Please note this PR depends on https://github.com/ezsystems/ezplatform-richtext/pull/150, and without schema changes, table-border edits won't be saved in the database. But it is still possible to test UI without that PR.

And some JavaScript improvement suggestions are expected here :)

**TODO**:
